contracts/protocol/integration/amm/ArrakisUniswapV3AmmAdapter.sol

```diff
@@ -0,0 +1,277 @@
+/*
+    Copyright 2022 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache-2.0
+*/
+
+pragma solidity 0.8.17;
+
+import "@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+import "@uniswap/v3-core/contracts/interfaces/IUniswapV3Factory.sol";
+
+import "../../../interfaces/IAmmAdapter.sol";
+import "../../../interfaces/external/IArrakisVaultV1.sol";
+
+/**
+ * @title UniswapV3AmmAdapter
+ * @author Zishan Sami
```

Comment:
Without diminishing Zishans contribution I'd suggest changing this to "Index Coop" as we usually do for all contracts I think.